File: apps/admin-car-rental/src/api/customer/CustomerCreateInput.ts
import { RentalCreateNestedManyWithoutCustomersInput } from "./RentalCreateNestedManyWithoutCustomersInput";
import { OrderCreateNestedManyWithoutCustomersInput } from "./OrderCreateNestedManyWithoutCustomersInput";
import { FeedbackRatingCreateNestedManyWithoutCustomersInput } from "./FeedbackRatingCreateNestedManyWithoutCustomersInput";

export type CustomerCreateInput = {
  rentals?: RentalCreateNestedManyWithoutCustomersInput;
  email?: string | null;
  licenseNumber?: string | null;
  lastName?: string | null;
  phone?: string | null;
  address?: string | null;
  firstName?: string | null;
  orders?: OrderCreateNestedManyWithoutCustomersInput;
  feedbackRatings?: FeedbackRatingCreateNestedManyWithoutCustomersInput;
};

File: apps/server-car-rental/src/APIs/Customer/Dtos/CustomerCreateInput.cs
namespace CarRentalService.APIs.Dtos;

public class CustomerCreateInput
{
    public DateTime CreatedAt { get; set; }
    public ICollection<RentalDto>? Rentals { get; set; }
    public string? LicenseNumber { get; set; }
    public string? LastName { get; set; }
    public string? Phone { get; set; }
    public string? Address { get; set; }
    public string? FirstName { get; set; }
}
